### PR TITLE
cmake: make enclave signature reproducible

### PR DESCRIPTION
**Pull request by rdzhou**
## Description

Make enclave signature reproducible regardless of the src/build

## Type of change (select applied and DELETE the others)

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
yes

### CODE REVIEW

File: cmake/scripts/cc_wrapper.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+set -e
+REQUIRED_ENVS=("MESATEE_PROJECT_ROOT" "MESATEE_BUILD_ROOT" "CMAKE_C_COMPILER")
+for var in "${REQUIRED_ENVS[@]}"; do
+    [ -z "${!var}" ] && echo "Please set ${var}" && exit -1
+done
+
+exec "${CMAKE_C_COMPILER}" "$@" -fdebug-prefix-map=${MESATEE_PROJECT_ROOT}=/mesatee_src -fdebug-prefix-map=${MESATEE_BUILD_ROOT}=/mesatee_build
```
Comment (mssun) [inline cmake/scripts/cc_wrapper.sh]: Can you add a comment before this line saying that why do we need to replace `debug-prefix-map`.

File: cmake/scripts/rustc_wrapper.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+set -e
+REQUIRED_ENVS=("MESATEE_PROJECT_ROOT" "MESATEE_BUILD_ROOT")
+for var in "${REQUIRED_ENVS[@]}"; do
+    [ -z "${!var}" ] && echo "Please set ${var}" && exit -1
+done
+
+exec rustc "$@" --remap-path-prefix=${HOME}/.cargo=/cargo_home --remap-path-prefix=${MESATEE_PROJECT_ROOT}=/mesatee_src --remap-path-prefix=${MESATEE_BUILD_ROOT}=/mesatee_build
```
Comment (mssun) [inline cmake/scripts/rustc_wrapper.sh]: Can you add a comment before this line saying that why do we need to replace `debug-prefix-map`.

**Comment by rdzhou** [comment]
> How about the `Makefile`?
Currently the fix is quite CMake-style. I will fix Makefile in another commit in Makefile-style.

File: cmake/scripts/rustc_wrapper.sh (hunk 2 above)
Comment (rdzhou) [reply cmake/scripts/rustc_wrapper.sh]: done

File: cmake/scripts/cc_wrapper.sh (hunk 1 above)
Comment (rdzhou) [reply cmake/scripts/cc_wrapper.sh]: done

**Pull request merged**